CHANGELOG.md

```diff
@@ -16,6 +16,7 @@
 ### Internals
 * Changed the metrics timers to more precisely report in nanoseconds, instead of seconds. ([#3359](https://github.com/realm/realm-core/issues/3359))
 * Better performance when cloud query metrics are turned on, by not acquiring a backtrace on query serialization errors (permissions queries). ([#3361](https://github.com/realm/realm-core/issues/3361)).
+* Performance improved for queries comparing a constant value to a property over unary link path (eg: "someLink.Id == 42"). ([#3670](https://github.com/realm/realm-core/issues/3370))
```

Comment:
If it affects end users, it should go into enhancements - and it surely does :-)
Same goes for the enhancement just above this.

Reply:
I see enhancements as functional enhancements. If there is something noted in this section, then minor version number should be bumped. An this would not be the case for these improvements. Perhaps we are missing a section called "non-functional improvements".

Reply:
By definition, it's intended to be any user facing enhancements. Those can be non functional as well. I don't think we need to differentiate whether it's functional or non functional?

Reply:
I agree with Jørgen here, semvar enhancements relate to the API, not the users. The core API hasn't changed at all here.

Reply:
We have defined that "Internals" in the release notes are sections that other teams and end users can ignore reading (it's internal the the team developing the component) and in some repos we cut it out automatically from the final release note.

As for SemVer, it actually doesn't really describe things like this. At best it's a Minor upgrade as  "It MAY be incremented if substantial new functionality or improvements are introduced within the private code."  (See [SemVer](https://semver.org/#spec-item-7)). Where as the Patch update only refers to bug fixes. 
But let's not spend too much time going down this rabit hole. My main point was that it's relevant to propagate this info to other teams so it gets included in their release notes.